homeassistant/components/plex/__init__.py

```diff
@@ -80,29 +79,34 @@ async def async_browse_media(hass, media_content_type, media_content_id, platfor
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up the Plex component."""
-    hass.data.setdefault(
-        DOMAIN,
-        {SERVERS: {}, DISPATCHERS: {}, WEBSOCKETS: {}, PLATFORMS_COMPLETED: {}},
-    )
-
-    await async_setup_services(hass)
-
-    hass.http.register_view(PlexImageView())
-
-    gdm = hass.data[DOMAIN][GDM_SCANNER] = GDM()
+    gdm = GDM()
 
     def gdm_scan():
         _LOGGER.debug("Scanning for GDM clients")
         gdm.scan(scan_for_clients=True)
 
-    hass.data[DOMAIN][GDM_DEBOUNCER] = Debouncer[None](
+    debouncer = Debouncer[None](
         hass,
         _LOGGER,
         cooldown=10,
         immediate=True,
         function=gdm_scan,
     ).async_call
 
+    hass_data = PlexData(
+        servers={},
+        dispatchers={},
+        websockets={},
+        platforms_completed={},
+        gdm_scanner=gdm,
+        gdm_debouncer=debouncer,
+    )
+    hass.data.setdefault(DOMAIN, hass_data)
```

Comment:
Note: there is a slight change of logic here.

Previously, `gdm_scanner`/`gdm_debouncer` were declared after `async_setup_services`/`register_view`.
Now they are declared before to ensure that PlexData isn't missing initialization items.